### PR TITLE
Update GH Workflows

### PR DESCRIPTION
**Pull request by gusrasch**
# Description

NO IMPACT TO MAGNIV FUNCTIONALITY; CI ONLY
- Separates linting from pytest
- Adds a e2e test step by building [test repo](https://github.com/MagnivOrg/magniv-repo-for-testing)

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My code follows the style guidelines of this project

### CODE REVIEW

**Comment by gusrasch** [comment]
not needed per https://github.com/MagnivOrg/magniv-core/pull/42

File: .github/workflows/linting.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.9", "3.10"]
```
Comment (jzone3) [inline .github/workflows/linting.yaml:17]: Why did we take of 3.9 support?

File: .github/workflows/e2e-tests.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.9", "3.10"]
+
+    steps:
+      - uses: actions/checkout@master
+        with:
+          repository: MagnivOrg/magniv-repo-for-testing
```
Comment (jzone3) [inline .github/workflows/e2e-tests.yaml:22]: Will need to put more stuff in this repo, but fine for this commit ✅

**Review by jzone3 — COMMENTED** [review]
LGTM. Just one question about Python versions.

Once this is merged, could you just do some commit that tests it all on master?

File: .github/workflows/linting.yaml (hunk 1 above)
Comment (gusrasch) [reply .github/workflows/linting.yaml:17]: I didn't think it was relevant for specifically the linting - I let it continue to run with both 3.9/3.10 for the actual tests, but I don't think version matters for formatting. I am happy to add it back for consistency, though

File: .github/workflows/e2e-tests.yaml (hunk 2 above)
Comment (gusrasch) [reply .github/workflows/e2e-tests.yaml:22]: yes agreed, figured maybe we can do that via PRs to the other repo and run this same test there too to make sure we don't break this repo

Comment (gusrasch) [reply .github/workflows/e2e-tests.yaml:22]: Added this here if you can also take a look https://github.com/MagnivOrg/magniv-repo-for-testing/pull/1

**Comment by jzone3** [comment]
LGTM, ready to merge 🎉

**Pull request merged**